### ISSUE TITLE
Hammer.js and Template interaction causes NPE

### ISSUE THREAD

**Issue by mhevery** [angular/angular#723]
Assume this template
```
<ul>
  <li !if="exp" (tap)="doSomething()"></li>
<ul>
```

Here the `li` will get turned into:
```
<template>
  <li !if="exp" (tap)="doSomething()"></li>
</template>
```

Once the `View` of the `if` gets instantiated it will attempt to register a `tap` event on `Hammer`. The issue is that the element is a clone of the `<template>` and is not yet inserted into the main document for rendering. `Hammer` will than try to register a listener on the wrong document, which is null. 

(See [`hammer.js`](https://github.com/hammerjs/hammer.js/blob/52d5887386b7ae8aa125ca80facc91023bfac43e/hammer.js#L324)) 
```
function getWindowForElement(element) {
    var doc = element.ownerDocument;
    return (doc.defaultView || doc.parentWindow);
}
```

The issue is that the `doc.defaultView` is null since the `ownerDocument` is that of the `template` and not of the main document. 

### Proposed Fix

Use [`document.importNode(element, true)`](https://developer.mozilla.org/en-US/docs/Web/API/document.importNode) instead of `element.cloneNode(true)`

**Comment by vicb** [comment]
Note that this is solved with in the master branch of [hammer](https://github.com/hammerjs/hammer.js/blob/52a090e79bf23af5da4bece36535b66191f5a063/src/utils.js#L314) which is why the gesture example embed a custom build based on master - it is **not** part of the latest official release of hammer.

I guess it's even better if `importNode()` fixes it.

**Comment by mhevery** [comment]
Fixed by #724
